sonic-thermalctld/scripts/thermalctld

```diff
@@ -592,6 +592,8 @@ class TemperatureUpdater(logger.Logger):
             self._refresh_temperature_status(CHASSIS_INFO_KEY, thermal, index)
 
         for psu_index, psu in enumerate(self.chassis.get_all_psus()):
+            if not psu.get_presence():
```

Comment:
I am not sure that we should check presence here. "N/A" is good enough to tell user that the PSU thermal is not available at the moment, no matter it is because of PSU absence or hardware error. Maybe you should fix your test expectation. But if you insit the change, please consider following items:
1. Do we need check psu.get_power_good_status here? If PSU is present but not powered on, thermal sensor should not be available.
2. Do we need remove previous DB entry here?
3. For `show platform fan` and `show platform psu`, non presence PSU will be displayed, should we align it with this design?
4. For existing sonic-mgmt test, can we guarantee no test is affected?
5. Unit test is needed to cover the new change.

Reply:
@cytsao1 @Junchao-Mellanox how about get_all_psus() return only those PSUs which are present? In that case, "N/A" would indicate some issue with physical present PSU?

Reply:
Only returning present PSUs from `get_all_psus` would cause issues with inconsistent indexing throughout the code; when `enumerate` is used, numbering will always be sequential from `0`, and inserting a new PSU may cause numbering of existing PSUs to change.